pkg/operator/infrastructurecontroller/infra_controller.go

```diff
@@ -0,0 +1,79 @@
+package infrastructurecontroller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+type InfrastructureController struct {
+	operatorClient operatorv1helpers.OperatorClient
+	infraClient    configv1client.InfrastructuresGetter
+}
+
+// degradedConditionNames lists all supported condition types.
+var degradedConditionNames = []string{
+	"InfrastructureControllerDegraded",
+}
+
+func NewInfrastructureController(operatorClient operatorv1helpers.OperatorClient, infraClient configv1client.InfrastructuresGetter, infraInformer cache.SharedIndexInformer, recorder events.Recorder) factory.Controller {
```

Comment:
I think you want v1helpers.KubeInformersForNamespaces or whatever we called it.

Reply:
if you need to read the kube resources? we don't for now :)